Source/WebKit/WebProcess/UserContent/WebUserContentController.cpp

```diff
@@ -417,7 +418,12 @@ void WebUserContentController::addUserScriptInternal(InjectedBundleScriptWorld&
 
             auto& mainFrame = page.mainFrame();
             if (userScript.injectedFrames() == UserContentInjectedFrames::InjectInTopFrameOnly) {
-                mainFrame.injectUserScriptImmediately(world.coreWorld(), userScript);
+                for (auto* frame = &mainFrame; frame; frame = frame->tree().traverseNext()) {
```

Comment:
It is a bit inefficient to check every frame to see if it's the main frame (when we know only the first one is). Would be more efficient to write like so:
```
mainFrame.injectUserScriptImmediately(world.coreWorld(), userScript);
for (auto* frame = mainFrame.tree().traverseNext(); frame; frame = frame->tree().traverseNext())
    frame->loader().client().dispatchGlobalObjectAvailable(world.coreWorld());
```